File: Split_a_String_in_Balanced_Strings/Split_a_String_in_Balanced_Strings.go
package main

import (
	"fmt"
)

func main() {
	s := "RLRRLLRLRL"
	fmt.Println(balancedStringSplit(s))
}

func balancedStringSplit(s string) int {
	var count int
	var j int
	for _, i := range s {
		if i == 'L' {
			j += 1
		} else {
			j -= 1
		}
		if j == 0 {
			count += 1
		}

	}
	return count

	// for _, char := range s {
	// 	switch char {
	// 	case 'L':
	// 		j++
	// 	case 'R':
	// 		j--
	// 	}
	// 	if j == 0 {
	// 		count++
	// 	}
	// }
	// return count
}

File: Split_a_String_in_Balanced_Strings/Split_a_String_in_Balanced_Strings.py
# Balanced strings are those that have an equal quantity of 'L' and 'R' characters.

# Given a balanced string s, split it into some number of substrings such that:

# Each substring is balanced.
# Return the maximum number of balanced strings you can obtain.



s = "RLRRLLRLRL"
def balancedStringSplit(s):
    
    count = 0
    j = 0
    for i in s:
        if i == "L":
            j+=1
        else:
            j-=1
        
        if j == 0:
            count+=1

    print(count)

print(balancedStringSplit(s))









